### PR TITLE
kops: Allow kops jobs to be run outside Jenkins

### PR DESCRIPTION
**Pull request by zmerlynn**
This lets you do e.g.

```
KOPS_E2E_STATE_STORE=s3://your-state-store \
KOPS_E2E_CLUSTER_NAME=e2e.your.domain \
jobs/ci-kubernetes-e2e-kops-aws.sh
```

### CODE REVIEW

File: jenkins/e2e-image/kops-e2e-runner.sh [hunk 1]
```diff
@@ -53,9 +53,12 @@ if [[ "${KOPS_DEPLOY_LATEST_KUBE:-}" =~ ^[yY]$ ]]; then
   export E2E_OPT="${E2E_OPT} --kops-kubernetes-version ${KOPS_KUBE_RELEASE_URL}/${KOPS_KUBE_LATEST}"
 fi
 
-EXTERNAL_IP=$(curl -SsL -H 'Metadata-Flavor: Google' 'http://metadata.google.internal/computeMetadata/v1/instance/network-interfaces/0/access-configs/0/external-ip')
+EXTERNAL_IP=$(curl -SsL -H 'Metadata-Flavor: Google' 'http://metadata.google.internal/computeMetadata/v1/instance/network-interfaces/0/access-configs/0/external-ip' || true)
 if [[ -z "${EXTERNAL_IP}" ]]; then
   # Running outside GCE
+  echo
+  echo "WARNING: Getting external IP from instance metadata failed, assuming not running on GCE."
+  echo
```
Comment (krzyzacy) [inline jenkins/e2e-image/kops-e2e-runner.sh]: eh, why three echo here?

Comment (zmerlynn) [inline jenkins/e2e-image/kops-e2e-runner.sh]: So it stands out if someone sees it in the actual Jenkins logs (because it's weird there).

**Comment by zmerlynn** [comment]
https://github.com/kubernetes/kubernetes/pull/38786 for e2e-image testing

File: jenkins/e2e-image/kops-e2e-runner.sh (hunk 1 above)
Comment (zmerlynn) [inline jenkins/e2e-image/kops-e2e-runner.sh]: Actually, if I can keep this, I'd appreciate it. I can fix it in a follow-on if desired, or slap a 4th commit on that fixes it, but the e2e-image push and my existing testing has this in it.

Comment (krzyzacy) [inline jenkins/e2e-image/kops-e2e-runner.sh]: I'm just curious :-) That's fine!

**Comment by krzyzacy** [comment]
I guess I'll leave @fejta to stamp this, while wait for your test results.

**Comment by zmerlynn** [comment]
Tests are green. If it looks good, mind if we get this in so I can verify I didn't break the `kops` Jenkins jobs? (I tested these changes locally, but there's a change the on-Jenkins path is broken). (Out rest of week.)

**Comment by krzyzacy** [comment]
/lgtm

sure

**Pull request merged**